plutus-core/testlib/PlutusCore/Generators/PIR/Substitutions.hs

```diff
@@ -0,0 +1,312 @@
+-- editorconfig-checker-disable-file
+
+module PlutusCore.Generators.PIR.Substitutions where
```

Comment:
Maybe we should split out the unification stuff? the substitutions and tools for manipulating them seem potentially independently useful, e.g. could go in `Subst` (making it like the third way of doing substitution oy vey)

Reply:
> Maybe we should split out the unification stuff? 

Done.

> the substitutions and tools for manipulating them seem potentially independently useful, e.g. could go in Subst (making it like the third way of doing substitution oy vey)

Wrote

```
{- Note [Substitution in generators]
Generators define their own substitution algorithm. It's generally useful, however it's quite
inefficient, since calling 'renameVar' in the worker when handling bindings makes the whole
algorithm at least quadratic. Plus, it also very general, since it handles both nested and parallel
substitution.

So for now we've decided to keep it local to the generators and extract it into the main library
once there's a specific reason to do that.
-}
```

Let's keep it in the generators folder for now.